### ISSUE TITLE
C# cref or \ref doesn't work without full namespace, even for classes in same namespace

### ISSUE THREAD

**Issue by runevision** [doxygen/doxygen#10766]
**Describe the bug**
Referencing another class (in this case called TopLayerDependency) in the same namespace with `<see cref="TopLayerDependency"/>` does not create a link when the class name is not qualified with the (common) namespace. In the output I see
`unable to resolve reference to 'TopLayerDependency' for \ref command`.

**Screenshots**

![MissingLink](https://github.com/doxygen/doxygen/assets/6685642/30e12cea-9f31-4602-99e6-5a88a7bae11c)

**To Reproduce**
Default Doxyfile without modifications used to reproduce the issue.

Place these two files in the folder Doxygen is run from:

File GenerationSource.cs
```
namespace Runevision.LayerGen {
    /// <summary>
    /// Unity component that creates a <see cref="TopLayerDependency"/>. <br/>
    /// Unity component that creates a \ref TopLayerDependency. <br/>
    /// Unity component that creates a <see cref="Runevision.LayerGen.TopLayerDependency"/>. <br/>
    /// Unity component that creates a \ref Runevision.LayerGen.TopLayerDependency.
    /// </summary>
    public class GenerationSource : MonoBehaviour { }
}
```

File TopLayerDependency.cs
```
namespace Runevision.LayerGen {
    /// <summary>
    /// All generation starts with one or more top level dependencies at the root.
    /// </summary>
    public class TopLayerDependency { }
}
```

**Expected behavior**
I would expect class names in the same namespace to be turned into links without having to specify the namespace.

**Version**
1.10.0 (GIT-NOTFOUND)
Doxygen installed via brew
macOS 12.5.1

**Additional context**
None of my other cross references get turned into links either (for example references from one method to another method in the same class). It seems like something is very broken about Doxygen reference resolution for xmldoc and/or C#, but I think this report here is the simplest demonstration, so let's start with that. With any luck, the fix for this will fix the other ones too.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10795

**Comment by zhangzq** [comment]
The same problems exist in Python. I need to write whole path, i.e. package.sub_package.file_name.class_name.function_name to resolve the function.

**Comment by albert-github** [comment]
@zhangzq The python problem should also be fixed with the patch #10795, did you try it?
Otherwise:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by zhangzq** [comment]
> @zhangzq The python problem should also be fixed with the patch #10795, did you try it? Otherwise:
> 
> * Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
> * Please also specify the **full** doxygen version used (`doxygen -v`).

it works. very cool! thanks.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

As the code has been reworked, can you please retest.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
